website/blog/2024-12-24-the-rise-and-evolution-of-rag-in-2024-a-year-in-review/index.md

```diff
@@ -0,0 +1,296 @@
+
+Therefore, the following landmark events revolve around the technical challenges of RAG:
+
+- The rise of multimodal document parsing tools.
+- The emergence of BM25 and hybrid search, rendering pure vector databases unnecessary as a separate category.
+
+### RAGFlow: Pioneering Innovations in RAG Technology
+
+On April 1, 2024, we open-sourced the complete RAG engine, RAGFlow, which has since garnered over 26,000 stars on GitHub by the end of the year. The initial two design highlights of RAGFlow have become universal design principles for RAG architecture:
+
+First, while naive RAG systems only provided text-based chunking tools, RAGFlow introduced a semantic chunking step for unstructured data to ensure the input data quality. This involves using specially trained models to parse document layouts, avoiding the interference caused by simple text chunking tools on different data layouts. As the open-source community increasingly uses these models to parse various documents, this approach has gained widespread acceptance.
+
+Second, from the outset, we firmly adopted enterprise-level search engines to provide hybrid search as the sole back-end database. By leveraging full-text search with BM25 capabilities, we ensured precise query performance. Although BM25 is nearly thirty years old, RAG has revitalized this classic technology. This year, many vector databases began offering BM25; notably, the well-known vector database Qdrant even proposed an improved version called BM42, which later turned out to be a mistake. Despite many databases claiming to support BM25, very few truly meet RAG's basic requirements. However, the rise of BM25 is undeniable; pure vector databases no longer need to exist as a separate category, as the concept of hybrid search has gained widespread acceptance.
+
+RAGFlow can be considered one of the key drivers behind these two events.
+
+### The Rise of  GraphRAG
+
+Microsoft's mid-year open-sourcing of GraphRAG was a groundbreaking event. As a library rather than an end-to-end solution, GraphRAG's rapid rise in popularity underscores its ability to tackle key issues with Retrieval-Augmented Generation (RAG), particularly the semantic gap. This issue has long been a challenge for search system developers, as queries and answers often fail to align perfectly. When search systems evolved into RAG models, this problem was amplified: while traditional search queries are defined by a few keywords, RAG queries are user questions. The shift from keywords to questions makes user intent even harder to discern, thereby exacerbating this semantic gap. GraphRAG is one design aimed at bridging this gap.
+
+### The Emergence of Latency Interaction Models Like Col-xxx vs. Multimodal RAG built on VLM and late Interaction Models
```

Comment:
The Emergence of Latency Interaction Models Like Col-xxx vs. Multimodal RAG built on VLM and late Interaction Models 
belongs to two key events respectively. 

### The Emergence of Late Interaction Models Like Col-xxx
### Multimodal RAG built on VLM and late Interaction Models